app/src/main/java/org/schabi/newpipe/local/history/StatisticsPlaylistFragment.java

```diff
@@ -96,14 +97,24 @@ public View onCreateView(@NonNull LayoutInflater inflater,
         return inflater.inflate(R.layout.fragment_playlist, container, false);
     }
 
+    @Override
+    public void setUserVisibleHint(boolean isVisibleToUser) {
+        super.setUserVisibleHint(isVisibleToUser);
+        if (activity != null && isVisibleToUser) {
+            setTitle(activity.getString(R.string.title_activity_history));
+        }
+    }
+
     ///////////////////////////////////////////////////////////////////////////
     // Fragment LifeCycle - Views
     ///////////////////////////////////////////////////////////////////////////
 
     @Override
     protected void initViews(View rootView, Bundle savedInstanceState) {
         super.initViews(rootView, savedInstanceState);
-        setTitle(getString(R.string.title_last_played));
+        if(!useAsFrontPage) {
```

Comment:
Redundant code: Same as with mIsVisibleToUser. This should maybe be put into `BaseFrament`.